CRM/PivotData/AbstractData.php

```diff
@@ -177,6 +187,51 @@ protected function replaceCustomLabels($label) {
     return $label;
   }
 
+  /**
+   * Loads data for the given custom field.
+   *
+   * @param array $field
+   *   Array with the field's data, with the field's name and the field's group
+   *
+   * @return array
+   */
+  protected function loadCustomFieldData($field) {
+    if (empty($field['name']) || empty($field['group'])) {
+      return array();
+    }
+
+    $sql = '
+      SELECT g.id AS group_id, f.id AS id, f.label AS label, f.data_type AS data_type, 
```

Comment:
you have the custom field name and the custom group name, so u can just use the API to get the info you want.